qCC/ccCommandLineCommands.h

```diff
@@ -1300,6 +1302,105 @@ struct CommandDropGlobalShift : public ccCommandLineInterface::Command
 	}
 };
 
+struct CommandSFColorScale : public ccCommandLineInterface::Command
+{
+	CommandSFColorScale() : ccCommandLineInterface::Command("SF color scale", COMMAND_SF_COLOR_SCALE) {}
+
+	virtual bool process(ccCommandLineInterface& cmd) override
+	{
+		cmd.print("[SF COLOR SCALE]");
+
+		if (cmd.arguments().empty())
+			return cmd.error(QString("Missing parameter: color scale file after \"-%1\"").arg(COMMAND_SF_COLOR_SCALE));
+
+		QString filename = cmd.arguments().takeFirst();
+
+		ccColorScale::Shared scale = ccColorScale::LoadFromXML(filename);
+
+		if (!scale)
+			return cmd.error(QString("Failed to read color scale file '%1'!").arg(filename));
+
+		if (cmd.clouds().empty())
+			return cmd.error(QString("No point cloud on which to change the SF color scale! (be sure to open one with \"-%1 [cloud filename]\" before \"-%2\")").arg(COMMAND_OPEN, COMMAND_SF_COLOR_SCALE));
+
+		for (size_t i = 0; i < cmd.clouds().size(); ++i)
+		{
+			ccScalarField* sf = static_cast<ccScalarField*>(cmd.clouds()[i].pc->getCurrentOutScalarField());
+			if (sf)
+			{
+				sf->setColorScale(scale);
+			}
+		}
+
+		if (cmd.autoSaveMode() && !cmd.saveClouds("COLOR_SCALE"))
+			return false;
+
+		return true;
+	}
+};
+
+struct CommandSFConvertToRGB : public ccCommandLineInterface::Command
+{
+	CommandSFConvertToRGB() : ccCommandLineInterface::Command("SF convert to RGB", COMMAND_SF_CONVERT_TO_RGB) {}
+
+	virtual bool process(ccCommandLineInterface& cmd) override
+	{
+		cmd.print("[SF CONVERT TO RGB]");
+
+		if (cmd.arguments().empty())
+			return cmd.error(QString("Missing parameter: boolean (whether to mix with existing colors or not) after \"-%1\"").arg(COMMAND_SF_CONVERT_TO_RGB));
+
+		QString mixWithExistingColorsStr = cmd.arguments().takeFirst().toUpper();
+		bool mixWithExistingColors = false;
+		if (mixWithExistingColorsStr == "TRUE")
+		{
+			mixWithExistingColors = true;
+		}
+		else if (mixWithExistingColorsStr != "FALSE")
+		{
+			return cmd.error(QString("Invalid boolean value after \"-%1\". Got '%2' instead of TRUE or FALSE.").arg(COMMAND_SF_CONVERT_TO_RGB, mixWithExistingColorsStr));
+		}
+
+		if (cmd.clouds().empty())
+			return cmd.error(QString("No point cloud on which to convert SF to RGB! (be sure to open one with \"-%1 [cloud filename]\" before \"-%2\")").arg(COMMAND_OPEN, COMMAND_SF_CONVERT_TO_RGB));
+
+		for (size_t i = 0; i < cmd.clouds().size(); ++i)
+		{
+			ccPointCloud* pc = cmd.clouds()[i].pc;
+			unsigned sfCount = pc->getNumberOfScalarFields();
+			int activeSFIndex = pc->getCurrentOutScalarFieldIndex();
```

Comment:
I found that 'SET_ACTIVE_SF' doesn't set the displayed scalar field and `setRGBColorWithCurrentScalarField` checks for a displayed field using `hasDisplayedScalarField`. To solve this, I just used `getCurrentOutScalarFieldIndex` and `setCurrentDisplayedScalarField` to set the active scalar field as the displayed scalar field.